features/step_definitions/common_steps.rb

```diff
@@ -405,18 +405,12 @@
 end
 
 Then /^I (don't |)see '(.*?)'(?: or '(.*)')? text on the page/ do |negative, expected_text, alternative_expected_text|
-  has_text = page.has_content?(expected_text)
+  method = negative.empty? ? :has_content? : :has_no_content?
 
-  has_alternative_text = false
-  if alternative_expected_text
-    has_alternative_text = page.has_content?(alternative_expected_text)
-  end
+  result = page.send(method, expected_text)
+  alternative_result = alternative_expected_text ? page.send(method, alternative_expected_text) : false
```

Comment:
Neat logic changes here, well done. I would only check with @brenetic style-wise about the use of ternary operators.